### PR TITLE
Add support for a Usabilla feedback button on every page

### PR DESCRIPTION
**Pull request by joelverhagen**
This is behind feature flag `NuGetGallery.UsabillaEveryPage`.

Progress on https://github.com/NuGet/NuGetGallery/issues/7385.

image

The survey questions can be changed from the Usabilla portal.

image

Note that when the modal opens and the page content does not fill the viewport (i.e. there is no vertical scroll bar, the footer jumps up). I have not found a workaround and there is similar behavior on other websites:

image

### CODE REVIEW

File: src/NuGetGallery/App_Code/ViewHelpers.cshtml [hunk 1]
```diff
@@ -351,6 +351,58 @@
...
+{
+    var config = DependencyResolver.Current.GetService<IGalleryConfigurationService>();
+    if (config != null)
+    {
+        var buttonId = config.Current.UsabillaFeedbackButtonId;
+        if (!string.IsNullOrWhiteSpace(buttonId))
+        {
+            var featureFlagService = DependencyResolver.Current.GetService<IFeatureFlagService>();
+            if (featureFlagService.IsUsabillaButtonEnabledOnEveryPage())
+            {
```
Comment (zhhyu) [inline src/NuGetGallery/App_Code/ViewHelpers.cshtml]: Maybe a shared method to do the check?

File: src/NuGetGallery.Services/Configuration/FeatureFlagService.cs [hunk 2]
```diff
@@ -33,6 +33,7 @@ public class FeatureFlagService : IFeatureFlagService
         private const string ODataCacheDurationsFeatureName = GalleryPrefix + "ODataCacheDurations";
         private const string ShowEnable2FADialog = GalleryPrefix + "ShowEnable2FADialog";
         private const string Get2FADismissFeedback = GalleryPrefix + "Get2FADismissFeedback";
+        private const string UsabillaEveryPage = GalleryPrefix + "UsabillaEveryPage";
```
Comment (zhhyu) [inline src/NuGetGallery.Services/Configuration/FeatureFlagService.cs]: UsabillaOnEveryPageFeatureName?

File: src/NuGetGallery/App_Code/ViewHelpers.cshtml (hunk 1 above)
Comment (joelverhagen) [reply src/NuGetGallery/App_Code/ViewHelpers.cshtml]: Good point! I'll do that.

File: src/NuGetGallery.Services/Configuration/FeatureFlagService.cs (hunk 2 above)
Comment (joelverhagen) [reply src/NuGetGallery.Services/Configuration/FeatureFlagService.cs]: Sure, sounds fine.

File: src/NuGetGallery/App_Data/Files/Content/flags.json [hunk 3]
```diff
@@ -7,7 +7,8 @@
     "NuGetGallery.GravatarProxyEnSubdomain": "Enabled",
     "NuGetGallery.ODataCacheDurations": "Enabled",
     "NuGetGallery.ShowEnable2FADialog": "Enabled",
-    "NuGetGallery.Get2FADismissFeedback": "Disabled"
+    "NuGetGallery.Get2FADismissFeedback": "Disabled",
+    "NuGetGallery.UsabillaEveryPage": "Enabled"
```
Comment (zhhyu) [inline src/NuGetGallery/App_Data/Files/Content/flags.json]: Should this match the new name?

Comment (joelverhagen) [reply src/NuGetGallery/App_Data/Files/Content/flags.json]: The name is the same. I just changed the constant name.

**Pull request merged**